src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -142,10 +155,30 @@ QString longname_from(const QFileInfo& file_info)
     return buff;
 }
 
-// Convert passed in octal to same hex digits for Qt Permissions flags
-auto convert_permissions(int perms)
+auto to_qt_permissions(int perms)
 {
-    return (QFileDevice::Permissions)(QString::number(perms & 07777, 8).toUInt(nullptr, 16));
+    QFileDevice::Permissions out;
+
+    if (perms & Permissions::read_user)
+        out |= QFileDevice::ReadUser;
+    if (perms & Permissions::write_user)
+        out |= QFileDevice::WriteUser;
+    if (perms & Permissions::exec_user)
+        out |= QFileDevice::ExeUser;
+    if (perms & Permissions::read_group)
+        out |= QFileDevice::ReadGroup;
+    if (perms & Permissions::write_group)
+        out |= QFileDevice::WriteGroup;
+    if (perms & Permissions::exec_group)
+        out |= QFileDevice::ExeGroup;
+    if (perms & Permissions::read_other)
+        out |= QFileDevice::ReadOther;
+    if (perms & Permissions::write_other)
+        out |= QFileDevice::WriteOther;
+    if (perms & Permissions::exec_other)
+        out |= QFileDevice::ExeOther;
+
+    return out;
 }
```

Comment:
On line 401, there is a call to go the other way- from Qt to the POSIX octal format.  We might as well do the explicit mapping for it as well.